homeassistant/components/mqtt/__init__.py

```diff
@@ -623,11 +631,23 @@ def __init__(self, hass: HomeAssistantType, config_entry, conf,) -> None:
         self.conf = conf
         self.subscriptions: List[Subscription] = []
         self.connected = False
+        self._ha_started = asyncio.Event()
+        self._last_subscribe = time.time()
         self._mqttc: mqtt.Client = None
         self._paho_lock = asyncio.Lock()
 
         self._pending_operations = {}
 
+        if self.hass.state == CoreState.running:
+            self._ha_started.set()
```

Comment:
We try to avoid side effects in init method and use a dedicated setup method for those.